module-4/backend/week_two.md

```diff
@@ -5,18 +5,55 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's one difference between ES5 and ES6?
+
+fat arrow functions. using let and const.
+
 2. What's the difference between asynchronous and synchronous JavaScript? 
+
+synchronous means everything happens in a fixed order where asychronous means that events can happen any order.
+
 3. What are the four pillars of Object Oriented programming?
+
+abstraction, encapsulation, inheritance and polymorphism
+
 4. What are some tools available in JavaScript to help you write object oriented code?
+
+constructors and prototypes
+
 5. What are some key concepts to be aware of when refactoring your JavaScript?
+
+breaking things out in the smaller functions, drying out your code so you arent writing the same thing over and over again.
+
 6. What's a callback function and what are some reasons when we use/need callback functions?
+
+its when you can pass a function inside of another function as an argument.
+
 7. What are the different scopes of variables in Javascript? When would you want to define global variables?
+
+you can have a variable inside of a function that will change as you go. and you can have a global variable set outside of the function that wont change as you you go.
```

Comment:
I'm a little unclear on what you mean since this is so short and I want to make sure you're clear that **global variables** can be accessed anywhere (and can be reassigned anywhere). We very rarely want to define global variables as they put us at risk for naming conflicts and are never garbage collected. **Variables scoped to a function** can only be accessed inside that function - they may or may not be changed/re-assigned, just depends on what you need to do with them.